### DISCUSSION TITLE
ignoreTags_OnUpdate ...can someone explain this better?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [StuffAnThings/qbit_manage#386] [Q&A] [answered]
I've read this 500 times, but I still have no idea what this does:

> ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has at least one or more of the tags defined here. Otherwise torrents will not be tagged if tags exist.
>     - noHL
>     - issue
>     - cross-seed

So... This is called "ignore tags on update" ...but then it doesn't ignore the tags you list to ignore, but then tags them anyways, and then also doesn't... .... .... what??? haha

What I would expect to happen is that any torrent with one of the tags on that list should be ignore on tag updates. Done. Meaning... if I added "WTF" to the tag list under `- noHL`, and I had some torrents with the tag WTF, qbm should just ignore that torrent, not change ANY tags on it at all, and then move on to the next one until it finds a torrent that doesn't have a tag on the ignore list.

As of right now, I can't figure out how to exclude a given set of torrents from having any tags updated. Or maybe there's a better way to do this that I have yet to figure out?

Bottom line, PTP has been offline for a long time. I want to keep their torrents seeding, but I want qbm to stop tagging them `no_tracker`. I put them in their own category, and even made some new PTP-specific tags to help me filter them out. But they keep getting tagged with `no_tracker` anyway. 

Help?

**Comment by bakerboy448** [comment]
> What I would expect to happen is that any torrent with one of the tags on that list should be ignore on tag updates.

no. Where does it say or imply it does this?
Did you not read the explanation you quoted? It explicitly says torrents with these tags are NOT ignored. 

From the config sample

> ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has at least one or more of the tags defined here. Otherwise torrents will not be tagged if tags exist.

from the docs
> When running --tag-update function, it will update torrent tags for a given torrent even if the torrent has one or more of the tags defined here.

from the docs for what tag update does

> Use this if you would like to update your tags and/or set seed goals/limit upload speed by tag. (Only adds tags to untagged torrents)

so the function by default only adds tags.  The ignore tags allows those tags to be added/removed/updated.   No where does anything anywhere say that those ignore tags are not processed and I'm very confused how you came to that conclusion.

>  I can't figure out how to exclude a given set of torrents from having any tags updated. 

you don't have share limits do anything with those tags.

for your specific issue you can't selectively run commands on a subset of torrents. Don't run the tag command then. The output from QBm is valid and your PTP torrents are being tagged correctly and as expected.

lastly by defintion private torrents with no tracker associated with the torrent cannot be seeded because there is no way to get peers

there seems to be no issue here other than you not wanting your torrents accurately tagged for some strange reason because PTP is down and gone (at best temporarily)

**Comment by SawkeeReemo** [comment]
> 
> > What I would expect to happen is that any torrent with one of the tags on that list should be ignore on tag updates.
> 
> no. Where does it say or imply it does this? Did you not read the explanation you quoted? It explicitly says torrents with these tags are NOT ignored.

Seriously, I really do appreciate your help and hard work on all this, but your delivery is just ••chef's kiss•• sometimes. 🤣 
Did I not specifically say "I've read this 500 times, but I still have no idea what this does" on a post with the subject "can someone please explain this better?"

Yes, I read it and I know what `tag_update` does. But I don't understand the purpose of `ignoreTags_OnUpdate` because the commented definition is not clear as written. 

By this wording, if you have `tag_update` turned on, all your tags defined throughout this config will update, regardless of whatever it is on this list even though the function is literally called `ignoreTags_OnUpdate`. How is the NOT confusing? 🤣  You add tags to a list called `ignore` for them not to be ignored? What?? This needs to be better defined. I still don't know what the intended purpose is.

> 
> so the function by default only adds tags. The ignore tags allows those tags to be added/removed/updated. No where does anything anywhere say that those ignore tags are not processed and I'm very confused how you came to that conclusion.
> 

Can you give me a specific example showing how this is meant to work? I don't understand how something called `ignore` actually does the opposite of ignoring somehow. I think a specific scenario might shed some light. And then maybe I could offer a suggestion for a definition that helps clarify, if you're open to it.


> > I can't figure out how to exclude a given set of torrents from having any tags updated.
> 
> you don't have share limits do anything with those tags.

I'm not sure how this applies to my statement. But let me move forward...

> 
> for your specific issue you can't selectively run commands on a subset of torrents. Don't run the tag command then. The output from QBm is valid and your PTP torrents are being tagged correctly and as expected.
> 

So in conclusion, there's no way to do what I'm asking with this script. Got it. Thanks! I'll try to come up with another solution.


> lastly by defintion private torrents with no tracker associated with the torrent cannot be seeded because there is no way to get peers
> 
> there seems to be no issue here other than you not wanting your torrents accurately tagged for some strange reason because PTP is down and gone (at best temporarily)

This last bit is basically semantics, and I'm pretty sure you actually understand what I'm saying. But allow me to re-word my statement for accuracy: 

I don't want to delete my PTP torrents because I have hope it will come back and these will begin seeding again. But also, for qBT management, I was hoping to find a way to exclude them from being processed by qbm until PTP becomes active again. Not being able to exclude these torrents means that I have to manually remove all the PTP `no_tracker` tags before I can see a true count. I personally do not automate deletions, I just want them tagged so I can do it myself (mainly because of trackers randomly having issues from time to time that cause them to be tagged `no_tracker` when they are actually just temporarily offline). 

Know what I mean?

**Reply by bakerboy448** [reply, marked as answer]
Simply put. Tags are only processed once. Tags in the ignore list can be re-processed. E.g. noHL is in there as a torrent may be re hardlinked and one would not want the torrent to be permanently tagged as nohardlink.

There's little reason to change from the defaults as it is designed for transient tags.

Config wording and wiki PR updates welcome.


> But also, for qBT management, I was hoping to find a way to exclude them from being processed by qbm until PTP becomes active again. 

again, no such function exists nor has anyone requested a function. Open a feature request.

> I just want them tagged so I can do it myself (mainly because of trackers randomly having issues from time to time that cause them to be tagged no_tracker when they are actually just temporarily offline).

this doesn't make sense as you're removing a valid tag. If there is no tracker for the torrent or it's down whatever the applicable tag is...that's a fact at run time.  Don't like it - bitch at your trackers to be no unavailable

**Comment by SawkeeReemo** [comment]
> 
> Simply put. Tags are only processed once. Tags in the ignore list can be re-processed. E.g. noHL is in there as a torrent may be re hardlinked and one would not want the torrent to be permanently tagged as nohardlink.

**_^^^^ THIS! ^^^^_** Copy and paste that into the config comment for `ignoreTags_OnUpdate`! That's perfect. The name of the function is a little counter-intuitive, but if you literally copy/paste this entire quote starting from "simply put" through to "as nohardlink," that makes it so much easier to understand, in my opinion.


> 
> > But also, for qBT management, I was hoping to find a way to exclude them from being processed by qbm until PTP becomes active again.
> 
> again, no such function exists nor has anyone requested a function. Open a feature request.

Will do!

> 
> > I just want them tagged so I can do it myself (mainly because of trackers randomly having issues from time to time that cause them to be tagged no_tracker when they are actually just temporarily offline).
> 
> this doesn't make sense as you're removing a valid tag. If there is no tracker for the torrent or it's down whatever the applicable tag is...that's a fact at run time. Don't like it - bitch at your trackers to be no unavailable

Correct. I don't want to remove a valid tag, but what I want is an option to have better management for things I know/hope to be temporarily offline to just be truly ignored until I end that option. So, I will make a feature request.

Thanks again, man! Appreciate it.
